<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StartLiveDataDialog</class>
 <widget class="QDialog" name="StartLiveDataDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>988</width>
    <height>905</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Start Live Data Monitoring</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <item>
    <widget class="QSplitter" name="splitterMain">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="widgetLeft" native="true">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="widgetLeftLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="lblInstrument">
           <property name="text">
            <string>Instrument:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cmbInstrument"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="styleSheet">
          <string notr="true">QGroupBox { border: 1px solid gray;  border-radius: 4px; font-weight: bold; margin-top: 4px; margin-bottom: 4px; padding-top: 16px; }
QGroupBox::title { background-color: transparent;  subcontrol-position: top center;  padding-top:4px; padding-bottom:4px; } 
</string>
         </property>
         <property name="title">
          <string>Starting Time</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QRadioButton" name="radNow">
              <property name="toolTip">
               <string>Start collecting data from current time</string>
              </property>
              <property name="text">
               <string>Now</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radStartOfRun">
              <property name="toolTip">
               <string>Start processing data from the beginning of the current run</string>
              </property>
              <property name="text">
               <string>Start of Run</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QRadioButton" name="radAbsoluteTime">
              <property name="maximumSize">
               <size>
                <width>17</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Start collecting data from a specific time:</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="dateTimeEdit">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximum">
               <number>9999</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="minutes">
              <property name="text">
               <string>minutes ago</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layoutUpdateEvery">
         <item>
          <widget class="QLabel" name="lblUpdateEvery">
           <property name="text">
            <string>Update Every:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="edtUpdateEvery"/>
         </item>
         <item>
          <widget class="QLabel" name="lblSeconds">
           <property name="text">
            <string>seconds</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QGroupBox" name="grpProcess">
           <property name="styleSheet">
            <string notr="true">QGroupBox { border: 1px solid gray;  border-radius: 4px; font-weight: bold; margin-top: 4px; margin-bottom: 4px; padding-top: 16px; }
QGroupBox::title { background-color: transparent;  subcontrol-position: top center;  padding-top:4px; padding-bottom:4px; } 
</string>
           </property>
           <property name="title">
            <string>Processing</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QRadioButton" name="radProcessNone">
              <property name="toolTip">
               <string>Perform no processing on data chunks</string>
              </property>
              <property name="text">
               <string>No Processing</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radProcessAlgorithm">
              <property name="toolTip">
               <string>Process chunks with the algorithm selected on the right</string>
              </property>
              <property name="text">
               <string>Algorithm</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radProcessScript">
              <property name="toolTip">
               <string>Process chunks using the python script on the right</string>
              </property>
              <property name="text">
               <string>Python Script</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <spacer name="horizontalSpacer_9">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCheckBox" name="chkPreserveEvents">
           <property name="text">
            <string>Preserve Events</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_10">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="lblPreserveEventsWarning">
         <property name="text">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#ff0000;&quot;&gt;Warning:&lt;/span&gt;&lt;span style=&quot; font-size:10pt; color:#ff0000;&quot;&gt; using &amp;quot;Preserve Events&amp;quot; may make processing slow down as the run accumulates a large number of events!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <spacer name="horizontalSpacer_8">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="lblInstrument_2">
           <property name="text">
            <string>Accumulation Method:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cmbAccumulationMethod"/>
         </item>
         <item>
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QGroupBox" name="grpPostProcessing">
           <property name="styleSheet">
            <string notr="true">QGroupBox { border: 1px solid gray;  border-radius: 4px; font-weight: bold; margin-top: 4px; margin-bottom: 4px; padding-top: 16px; }
QGroupBox::title { background-color: transparent;  subcontrol-position: top center;  padding-top:4px; padding-bottom:4px; } 
</string>
           </property>
           <property name="title">
            <string>Post Processing</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QRadioButton" name="radPostProcessNone">
              <property name="toolTip">
               <string>Perform no processing on data chunks</string>
              </property>
              <property name="text">
               <string>No Processing</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radPostProcessAlgorithm">
              <property name="toolTip">
               <string>Process chunks with the algorithm selected on the right</string>
              </property>
              <property name="text">
               <string>Algorithm</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radPostProcessScript">
              <property name="toolTip">
               <string>Process chunks using the python script on the right</string>
              </property>
              <property name="text">
               <string>Python Script</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <spacer name="horizontalSpacer_11">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="lblRunTransitionBehavior">
           <property name="text">
            <string>At Run Transition:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cmbRunTransitionBehavior"/>
         </item>
         <item>
          <spacer name="horizontalSpacer_12">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="lblAccumulationWorkspace">
           <property name="text">
            <string>Accumulation Workspace: </string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="editAccumulationWorkspace"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Output Workspace: </string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="editOutputWorkspace"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="listenerProps">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">QGroupBox { border: 1px solid gray;  border-radius: 4px; font-weight: bold; margin-top: 4px; margin-bottom: 4px; padding-top: 16px; }
QGroupBox::title { background-color: transparent;  subcontrol-position: top center;  padding-top:4px; padding-bottom:4px; } 
</string>
         </property>
         <property name="title">
          <string>Listener Properties</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabProcessAlgo">
       <attribute name="title">
        <string>Processing Step</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayoutProcessing">
        <item>
         <widget class="MantidQt::MantidWidgets::ProcessingAlgoWidget" name="processingAlgo" native="true">
          <property name="editorVisible" stdset="0">
           <bool>false</bool>
          </property>
          <property name="infoString" stdset="0">
           <string>Select either Algorithm or Python Script under &quot;Processing&quot; to perform processing on each chunk of live data.
In the python script 'input' is a reference to the input workspace and 'output' is a string of the name of the output workspace.</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabPostAlgo">
       <attribute name="title">
        <string>Post-Processing Step</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayoutPostProcessing">
        <item>
         <widget class="MantidQt::MantidWidgets::ProcessingAlgoWidget" name="postAlgo" native="true">
          <property name="infoString" stdset="0">
           <string>Select either Algorithm or Python Script under &quot;Post Processing&quot; to perform processing on the accumulated data, after each chunk is added.
In the python script 'input' is a reference to the input workspace and 'output' is a string of the name of the output workspace.
           </string>
          </property>
          <property name="editorVisible" stdset="0">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MantidQt::MantidWidgets::ProcessingAlgoWidget</class>
   <extends>QWidget</extends>
   <header>MantidQtMantidWidgets/ProcessingAlgoWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
